test/integration/scheduler/taint_test.go

```diff
@@ -746,7 +749,7 @@ func TestTaintBasedEvictions(t *testing.T) {
 						select {
 						case <-heartbeatChans[i]:
 							return
-						case <-time.Tick(2 * time.Second):
+						case <-time.Tick(heartbeatInternal):
 							nodes[i].Status.Conditions = []v1.NodeCondition{
 								{
 									Type:              v1.NodeReady,
```

Comment:
instead of the logic you added below, can we just update the neededNode status here consistently?:

```
c := []v1.NodeCondition{{Type: v1.NodeReady, Status: v1.ConditionTrue, LastHeartbeatTime: metav1.Now(),},
 if nodes[i] == neededNode {
   c = test.nodeConditions
}
nodes[i].Status.Conditions = c
...
```

Reply:
Sounds good, let me update.